packages/sil-governance/src/routes/dashboard/components/ProposalsList.tsx

```diff
@@ -34,14 +48,47 @@ const getFilter = (filterType: ElectionFilter, currentUser: Address | null): Pro
   }
 };
 
-function compareById<E extends { readonly id: number }>(element1: E, element2: E): number {
-  return element1.id - element2.id;
-}
+type ProposalsComparator = (proposal1: ProposalProps, proposal2: ProposalProps) => number;
+
+const compareByIdDescending: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1 > proposal2 ? 1 : -1;
+};
+const compareByExpiryDate: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.expiryDate.getTime() - proposal2.expiryDate.getTime();
+};
+const compareByStartDate: ProposalsComparator = (proposal1, proposal2): number => {
+  return proposal1.startDate.getTime() - proposal2.startDate.getTime();
+};
+const compareByVote: ProposalsComparator = (proposal1, proposal2): number => {
+  if (proposal2.vote === undefined) return -1;
+  if (proposal1.vote === undefined) return 1;
```

Comment:
Same feedback as before: you need to handle the case `proposal1.vote === undefined && proposal2.vote === undefined` explicitely. Otherise `compareByVote(proposal1, proposal2) === -1` and `compareByVote(proposal2, proposal1) === -1` which means `proposal1 < proposal2` and `proposal2 < proposal1` at the same time, which is incorrect and will lead to wrong sorting